class.jetpack-options.php

```diff
@@ -134,32 +157,28 @@ public static function update_options( $array ) {
 	 * @param string|array $names
 	 */
 	public static function delete_option( $names ) {
+		$result = true;
 		$names = (array) $names;
 
-		foreach ( array_diff( $names, self::get_option_names(), self::get_option_names( 'non_compact' ) ) as $unknown_name ) {
+		foreach ( array_diff( $names, self::get_option_names(), self::get_option_names( 'non_compact' ), self::get_option_names( 'private' ) ) as $unknown_name ) {
```

Comment:
Might be useful to add a function that just checks if the option exists or not. Since we are doing this kind of checking in more then one place.